sdk/spring-sdk/src/main/scala/kalix/springsdk/impl/ActionDescriptorFactory.scala

```diff
@@ -61,64 +86,81 @@ private[impl] object ActionDescriptorFactory extends ComponentDescriptorFactory
         KalixMethod(SubscriptionServiceMethod(method))
           .withKalixOptions(kalixOptions)
       }
+      .toIndexedSeq
 
-    val subscriptionEventSourcedEntityMethods = component.getMethods
+    // methods annotated with @Subscribe.EventSourcedEntity
+    val subscriptionEventSourcedEntityMethods: IndexedSeq[KalixMethod] = component.getMethods
       .filter(hasEventSourcedEntitySubscription)
       .sorted // make sure we get the methods in deterministic order
       .map { method =>
-        if (hasEventSourcedEntitySubscription(component))
-          throw InvalidComponentException(
-            s"You cannot use Subscribe annotation in both the methods and the class. You can do either one or the other.")
         val subscriptionOptions = eventingInForEventSourcedEntity(method)
         val kalixOptions =
           kalix.MethodOptions.newBuilder().setEventing(subscriptionOptions).build()
-
         KalixMethod(SubscriptionServiceMethod(method))
           .withKalixOptions(kalixOptions)
       }
+      .toIndexedSeq
 
-    val subscriptionEventSourcedEntityClass =
+    val subscriptionEventSourcedEntityClass: Map[String, Seq[KalixMethod]] =
       if (hasEventSourcedEntitySubscription(component)) {
-        component.getMethods.sorted // make sure we get the methods in deterministic order
-          .filter(hasActionOutput)
-          .collect { case method =>
-            val subscriptionOptions = eventingInForEventSourcedEntity(component)
-            val kalixOptions =
-              kalix.MethodOptions.newBuilder().setEventing(subscriptionOptions).build()
+        val kalixMethods =
+          component.getMethods
+            .filter(hasActionOutput)
+            .sorted // make sure we get the methods in deterministic order
+            .map { method => KalixMethod(SubscriptionServiceMethod(method)) }
+            .toSeq
 
-            KalixMethod(SubscriptionServiceMethod(method))
-              .withKalixOptions(kalixOptions)
-          }
-      } else Array.empty[KalixMethod]
+        val entityType = findEventSourcedEntityType(component)
+        Map(entityType -> kalixMethods)
+
+      } else Map.empty
 
-    val subscriptionTopicMethods = component.getMethods
+    val subscriptionStreamClass: Map[String, Seq[KalixMethod]] = {
+      streamSubscription(component)
```

Comment:
This was missing. I mentioned today that subscribing with an action was not working. 

Well, the reason was that when use `@Subscribe.Stream`, the `ActionDescriptorFactory` won't generate any method in gRPC. It was only generating methods for `@Subscribe.EventSourcedEntity`, `@Subscribe.ValueEntity` and `@Subscribe.Topic`.

And the tests was not picking it because of a mistake and lack of some extra checks.

Reply:
Nice